File: app/src/main/java/mx/dev/franco/automusictagfixer/persistence/cache/CoverDataCache.java
package mx.dev.franco.automusictagfixer.persistence.cache;


import androidx.collection.ArrayMap;

import java.util.Map;

import mx.dev.franco.automusictagfixer.interfaces.Cache;

/**
 * Created by Franco Castillo on 13/04/2018.
 */
public class CoverDataCache implements Cache<String, byte[]> {
    private Map<String, byte[]> cache;

    public CoverDataCache() {
        cache = new ArrayMap<>();
    }

    @Override
    public void add(String key, byte[] value) {
        cache.put(key, value);
    }

    @Override
    public void delete(String key) {
        cache.remove(key);
    }

    @Override
    public byte[] load(String key) {
        return cache.get(key);
    }

    @Override
    public void deleteAll() {
        cache.clear();
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/covermanager/CoverDataCache.kt
package mx.dev.franco.automusictagfixer.covermanager

import androidx.collection.ArrayMap
import mx.dev.franco.automusictagfixer.interfaces.Cache

/**
 * Created by Franco Castillo on 13/04/2018.
 */
class CoverDataCache : Cache<String?, ByteArray?> {
    private val cache = ArrayMap<String?, ByteArray?>()
    override fun add(key: String?, value: ByteArray?) {
        cache[key] = value
    }

    override fun delete(key: String?) {
        cache.remove(key)
    }


    override fun load(key: String?) = cache[key]


    override fun deleteAll() = cache.clear()
}